### PR TITLE
Use the -test.timeout flag for go testing rather than a subprocess timeout

### PR DESCRIPTION
**Pull request by mikekap**
This gives a slightly nicer output which includes all the active goroutine stacks.

### CODE REVIEW

File: src/com/facebook/buck/go/GoTest.java [hunk 1]
```diff
@@ -107,11 +107,12 @@ public boolean hasTestResultFiles(ExecutionContext executionContext) {
...
                 .addAll(testMain.getExecutableCommand().getCommandPrefix(getResolver()))
                 .add("-test.v")
-                .add("-test.timeout", executionContext.getDefaultTestTimeoutMillis() + "ms")
+                .add(
+                    "-test.timeout",
+                    testRuleTimeoutMs.or(executionContext.getDefaultTestTimeoutMillis()) + "ms")
                 .build(),
             testMain.getExecutableCommand().getEnvironment(getResolver()),
             getPathToTestExitCode(),
-            testRuleTimeoutMs,
```
Comment (sdwilsh) [inline src/com/facebook/buck/go/GoTest.java]: What about adding some padding here and using both (in case the executable does hang)?

**Comment by sdwilsh** [comment]
@facebook-github-bot shipit

File: src/com/facebook/buck/go/GoTest.java [hunk 2]
```diff
@@ -37,6 +37,7 @@
 import com.facebook.buck.test.TestRunningOptions;
 import com.facebook.buck.test.result.type.ResultType;
 import com.google.common.base.Charsets;
+import com.google.common.base.Function;
```
Comment (sdwilsh) [inline src/com/facebook/buck/go/GoTest.java]: I should have looked at the travis build.  This is an unused import, so `ant lint` fails.

**Comment by sdwilsh** [comment]
@facebook-github-bot shipit

File: test/com/facebook/buck/go/GoTestIntegrationTest.java [hunk 3]
```diff
@@ -66,9 +65,7 @@ public void testGoTest() throws IOException {
   @Test
   public void testGoTestTimeout() throws IOException {
     ProjectWorkspace.ProcessResult result = workspace.runBuckCommand("test", "//:test-spinning");
-    result.assertSpecialExitCode("test should fail", 42);
-    String stderr = result.getStderr();
-    assertTrue(stderr, stderr.contains("Timed out running test command"));
+    result.assertTestFailure("test timed out");
```
Comment (sdwilsh) [inline test/com/facebook/buck/go/GoTestIntegrationTest.java]: @dreiss just pushed 7c2311282079c9e06117974b1d025b0079452d19 which conflicts with this, and I want to make sure that the end result should still be the same here.

Comment (mikekap) [inline test/com/facebook/buck/go/GoTestIntegrationTest.java]: Updated to include the timeout in the assert (go's testing does this by default)

**Pull request closed without merging**